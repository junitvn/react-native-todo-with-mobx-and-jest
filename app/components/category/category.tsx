import * as React from "react"
import { StyleProp, TextStyle, View, ViewStyle } from "react-native"
import { observer } from "mobx-react-lite"
import { color, spacing, typography } from "../../theme"
import { Text } from "../text/text"
import { Category as CategoryModel } from "../../models"
import { ProgressBar } from "../progress-bar/progress-bar"

const CONTAINER: ViewStyle = {
  justifyContent: "space-evenly",
  height: 120,
  width: 200,
  backgroundColor: "white",
  marginRight: spacing[4],
  borderRadius: 20,
  padding: spacing[2],
}

const TEXT: TextStyle = {
  fontFamily: typography.primary,
  fontSize: 28,
  color: "black",
  fontWeight: "500",
}

const TEXT_ALL_TASKS: TextStyle = {
  fontFamily: typography.primary,
  fontSize: 15,
  color: "gray",
}

const PROGRESS: TextStyle = {
  marginTop: spacing[2],
}

export interface CategoryProps {
  style?: StyleProp<ViewStyle>
  category: CategoryModel
}

export const Category = observer(function Category(props: CategoryProps) {
  const { style, category } = props
  const { title, completedTasks, tasks, color } = category
  const styles = Object.assign({}, CONTAINER, style)

  return (
    <View style={styles}>
      <Text style={TEXT_ALL_TASKS}>40 tasks</Text>
      <Text style={TEXT}>{title}</Text>
      <ProgressBar style={PROGRESS} current={5} total={10} color={color} />
    </View>
  )
})
