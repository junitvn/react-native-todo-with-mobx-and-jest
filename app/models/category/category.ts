import { TaskModel } from "./../task/task"
import { Instance, SnapshotIn, SnapshotOut, types } from "mobx-state-tree"

/**
 * Model description here for TypeScript hints.
 */
export const CategoryModel = types.model("Category").props({
  id: types.identifierNumber,
  title: types.string,
  tasks: types.optional(types.array(TaskModel), []),
  completedTasks: types.optional(types.array(TaskModel), []),
  color: types.maybe(types.string)
})
// eslint-disable-line @typescript-eslint/no-unused-vars

export interface Category extends Instance<typeof CategoryModel> {}
export interface CategorySnapshotOut extends SnapshotOut<typeof CategoryModel> {}
export interface CategorySnapshotIn extends SnapshotIn<typeof CategoryModel> {}
export const createCategoryDefaultModel = () => types.optional(CategoryModel, {})
