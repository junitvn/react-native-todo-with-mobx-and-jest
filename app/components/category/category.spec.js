import React from "react"
import renderer from "react-test-renderer"
import { Category } from "./category"

test("renders correctly", () => {
  const tree = renderer.create(<Category category={{ title: "Test", color: "gray" }} />).toJSON()
  expect(tree).toMatchSnapshot()
})
