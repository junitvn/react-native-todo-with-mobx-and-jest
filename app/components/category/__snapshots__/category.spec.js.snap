// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "height": 120,
      "justifyContent": "space-evenly",
      "marginRight": 16,
      "padding": 8,
      "width": 200,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Helvetica",
          "fontSize": 15,
        },
        Object {
          "color": "gray",
          "fontFamily": "Helvetica",
          "fontSize": 15,
        },
      ]
    }
  >
    40 tasks
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontFamily": "Helvetica",
          "fontSize": 15,
        },
        Object {
          "color": "black",
          "fontFamily": "Helvetica",
          "fontSize": 28,
          "fontWeight": "500",
        },
      ]
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "lightgray",
        "borderRadius": 3,
        "height": 3,
        "justifyContent": "flex-start",
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "height": 3,
          },
          Object {
            "backgroundColor": "gray",
            "width": "50%",
          },
        ]
      }
    />
  </View>
</View>
`;
